f5/bigip/resource.py

```diff
@@ -295,6 +305,37 @@ def _handle_requests_params(self, kwargs):
             requests_params.update({'params': params})
         return requests_params
 
+    def _check_exclusive_parameters(self, **kwargs):
+        """Check for mutually exclusive attributes in kwargs.
+
+        :raises ExclusiveAttributesPresent
+        """
+        if len(self._meta_data['exclusive_attributes']) > 0:
+            attr_set = set(kwargs.keys())
+            ex_set = set(self._meta_data['exclusive_attributes'][0])
+            common_set = attr_set.intersection(ex_set)
+            if len(common_set) > 1:
+                cset = ', '.join(common_set)
+                error = 'Mutually exclusive arguments submitted. ' \
+                        'The following arguments cannot be set ' \
+                        'together: "%s".' % cset
+                raise ExclusiveAttributesPresent(error)
+
+    @staticmethod
+    def _check_required_parameters(rqset, **kwargs):
+        """Helper function to do operation on sets
+
+        ::returns bool and variable
+        """
+        key_set = set(kwargs.keys())
+        required_minus_received = rqset - key_set
+        if required_minus_received != set():
```

Comment:
I have done the change and it breaks all of the other tests...i will do something else and just make the method return true or false, and we raise exceptions based on usage.